services/cd-service/pkg/event/writer.go

```diff
@@ -0,0 +1,139 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+package event
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/util"
+)
+
+func write(
+	filesystem billy.Filesystem,
+	dir string,
+	val any,
+) error {
+	return writeFile(filesystem, dir, reflect.ValueOf(val), encodingDefault)
+}
+
+func writeFile(
+	fs billy.Filesystem,
+	file string,
+	value reflect.Value,
+	encoding encoding,
```

Comment:
why is `encoding` needed? It seems to be passed only to `writeFile` recursively.